astroquery/heasarc/core.py

```diff
@@ -81,5 +210,109 @@ def _parse_result(self, response, verbose=False):
+              'hh mm ss,dd mm ss'
+        fields : str, optional
+            Return format for columns from the server available options:
+            * Standard (default) : Return default table columns
+            * All                : Return all table columns
+            * <custom>           : User defined csv list of columns to be
+              returned
+        radius : float (arcmin), optional
+            Astropy Quantity object, or a string that can be parsed into one.
+            e.g., '1 degree' or 1*u.degree.
+        coordsys: str, optional
+            If 'entry' is a set of coordinates, this specifies the coordinate
+            system used to interpret them. By default, equatorial coordinates
+            are assumed. Possible values:
+            * 'fk5' <default> (FK5 J2000 equatorial coordinates)
+            * 'fk4'           (FK4 B1950 equatorial coordinates)
+            * 'equatorial'    (equatorial coordinates, `equinox` param
+              determines epoch)
+            * 'galactic'      (Galactic coordinates)
+        equinox : int, optional
+            Epoch by which to interpret supplied equatorial coordinates
+            (defaults to 2000, ignored if `coordsys` is not 'equatorial')
+        resultmax : int, optional
+            Set maximum query results to be returned
+        sortvar : str, optional
+            Set the name of the column by which to sort the results. By default
+            the results are sorted by distance from queried object/position
+
+        displaymode : str, optional
+            Return format from server. Since the user does not interact with
+            this directly, it's best to leave this alone
+        action : str, optional
+            Type of action to be taken (defaults to 'Query')
+        """
+        # Define the basic query for this object
+        request_payload = dict(
+            tablehead=('name=BATCHRETRIEVALCATALOG_2.0 {}'
+                       .format(kwargs.get('mission'))),
+            Entry=kwargs.get('entry', 'none'),
+            Action=kwargs.get('action', 'Query'),
+            displaymode=kwargs.get('displaymode', 'FitsDisplay')
+        )
+
+        # Fill in optional information for refined queries
+
+        # Handle queries involving coordinates
+        coordsys = kwargs.get('coordsys', 'fk5')
+        if coordsys.lower() == 'fk5':
+            request_payload['Coordinates'] = 'Equatorial: R.A. Dec'
+
+        elif coordsys.lower() == 'fk4':
+            request_payload['Coordinates'] = 'Equatorial: R.A. Dec'
+            request_payload['equinox'] = 1950
+
+        elif coordsys.lower() == 'equatorial':
+            request_payload['Coordinates'] = 'Equatorial: R.A. Dec'
+
+            equinox = kwargs.get('equinox', None)
+            if equinox is not None:
+                request_payload['Equinox'] = str(equinox)
+
+        elif coordsys.lower() == 'galactic':
+            request_payload['Coordinates'] = 'Galactic: LII BII'
```

Comment:
perhaps add an `else: raise ValueError("Invalid coordsys")` or something along these lines?

Reply:
Good suggestion! I've added a raise that provides a list of the acceptable parameters.